octree/basic_utils_test.go

```diff
@@ -0,0 +1,187 @@
+
+		test.That(t, basicOct.nodeType, test.ShouldResemble, InternalNode)
+		test.That(t, basicOct.point, test.ShouldResemble, pc.PointAndData{})
+		test.That(t, basicOct.tree, test.ShouldResemble, children)
+	})
+}
+
+// Tests that the splitting of a filled octree node results in seven empty leaf nodes
+// and one filled one or an empty octree node will result in eight empty children nodes.
+func TestSplitIntoOctants(t *testing.T) {
+	ctx := context.Background()
+	logger := golog.NewTestLogger(t)
+
+	center := r3.Vector{X: 0, Y: 0, Z: 0}
+	side := 1.0
+
+	t.Run("Splitting empty octree node into octants", func(t *testing.T) {
+		basicOct, err := createNewOctree(ctx, center, side, logger)
+		test.That(t, err, test.ShouldBeNil)
+
+		err = basicOct.splitIntoOctants()
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, len(basicOct.node.tree), test.ShouldEqual, 8)
+		test.That(t, basicOct.node.nodeType, test.ShouldResemble, InternalNode)
+		test.That(t, basicOct.node.point, test.ShouldResemble, pc.PointAndData{})
+
+		for _, child := range basicOct.node.tree {
+			test.That(t, child.node.nodeType, test.ShouldResemble, LeafNodeEmpty)
+			test.That(t, child.node.point, test.ShouldResemble, pc.PointAndData{})
+		}
+	})
+
+	t.Run("Splitting filled basic octree node into octants", func(t *testing.T) {
+		basicOct, err := createNewOctree(ctx, center, side, logger)
+		test.That(t, err, test.ShouldBeNil)
+
+		p := r3.Vector{X: 0, Y: 0, Z: 0}
+		d := pc.NewValueData(1.0)
+
+		err = basicOct.Set(p, d)
+		test.That(t, err, test.ShouldBeNil)
+
+		err = basicOct.splitIntoOctants()
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, len(basicOct.node.tree), test.ShouldEqual, 8)
+		test.That(t, basicOct.node.nodeType, test.ShouldResemble, InternalNode)
+		test.That(t, basicOct.node.point, test.ShouldResemble, pc.PointAndData{})
+
+		filledLeaves := []*basicOctree{}
+
+		for _, child := range basicOct.node.tree {
+			if child.node.nodeType == LeafNodeFilled {
+				filledLeaves = append(filledLeaves, child)
+			}
+		}
+		test.That(t, len(filledLeaves), test.ShouldEqual, 1)
+		test.That(t, filledLeaves[0].node.point, test.ShouldResemble, pc.PointAndData{P: p, D: d})
+	})
+
+	t.Run("Splitting partially filled basic octree node into octants", func(t *testing.T) {
+		basicOct, err := createNewOctree(ctx, center, side, logger)
+		test.That(t, err, test.ShouldBeNil)
+
+		p1 := r3.Vector{X: 0, Y: 0, Z: 0}
+		d1 := pc.NewValueData(1.0)
+		basicOct.Set(p1, d1)
+		test.That(t, err, test.ShouldBeNil)
+
+		p2 := r3.Vector{X: .5, Y: 0, Z: 0}
+		d2 := pc.NewValueData(2.0)
+		basicOct.Set(p2, d2)
+		test.That(t, err, test.ShouldBeNil)
+
+		d, ok := basicOct.At(p1.X, p1.Y, p1.Z)
+		test.That(t, ok, test.ShouldBeTrue)
+		test.That(t, d, test.ShouldResemble, d1)
+
+		d, ok = basicOct.At(p2.X, p2.Y, p2.Z)
+		test.That(t, ok, test.ShouldBeTrue)
+		test.That(t, d, test.ShouldResemble, d2)
+
+		_, ok = basicOct.At(0, 1, .5)
+		test.That(t, ok, test.ShouldBeFalse)
```

Comment:
[q] Why are you checking this in this test?

Reply:
Too confirm that splitIntoOctant still places points  in the correct location and that `At()` can find them. It is repeating tests done in `At()` but i'd prefer to have confirmation in both locations